### ISSUE TITLE
Document getUploadedFiles()

### ISSUE THREAD

**Issue by akrabat** [slimphp/Slim-Website#195]
See https://github.com/slimphp/Slim/issues/2030

**Comment by llvdl** [comment]
The `getUploadedFiles` function is documented, but some information is still missing:

* when posting a form, the attribute "enctype" needs to be set to "multipart/form-data"
* when multiple files are uploaded for the same input key, the name has to be suffixed with brackets in the HTML form, e.g. `<input type="file" name="files[]" multiple="muitiple"/>`

See also:

* http://discourse.slimframework.com/t/how-to-upload-multiple-files-with-getuploadedfiles/1055
* https://www.codecourse.com/forum/topics/slim-3-uploaded-files

**Comment by silentworks** [comment]
@llvdl I think this would be better in the cookbook section, because not everyone who uses Slim uses it with an html form. Some are using it as an API and I am not sure if this would be beneficial or confusing in this part of the documentation.

**Comment by geggleto** [comment]
I always hated UploadedFiles tbqh.

**Comment by llvdl** [comment]
Thanks for the feedback!

@silentworks, I moved the documentation on the HTML forms to a cookbook page and added link. That also allows for a more detailed explanation. After reading the section, I think a small example of Slim code that handles the uploaded files would also be a nice addition, I will try to add that to the PR tomorrow.

Regarding the usefulness to API users: Unless I misunderstand the PHP docs, `$_FILES` only contains files posted using a form. `getUploadedFiles` has no use then for those users.

@geggleto Maybe because you never got to read the missing documentation? :stuck_out_tongue_winking_eye:  Jokes aside, if you have an alternative for `getUploadedFiles` then I am willing to add it to the PR. It's just that I saw a question on the message board and figured the existing documentation can be improved upon.

Regards,

Lennaert
